scripts/protocgen.sh

```diff
@@ -1,39 +1,46 @@
 #!/usr/bin/env bash
+set -euo pipefail
 
-set -eo pipefail
+# By default, this script runs against the latest commit to the master branch
+# in the Tendermint spec repository. To use this script with a different version
+# of the spec repository, run it with the $VERS environment variable set to the
+# desired commit in the spec repo.
 
 : ${VERS:=master}
-URL_PATH=archive/
+URL_PATH=tarball/
 if [[ VERS -ne master ]]; then
-    URL_PATH=archive/refs/tags/v
+    URL_PATH=tarball/refs/tags/v
 fi
 
-# Edit this line to clone your branch, if you are modifying protobuf files
-curl -qL "https://github.com/tendermint/spec/${URL_PATH}${VERS}.tar.gz" | tar -xjf - spec-"$VERS"/proto/
+echo "fetching proto files"
 
-cp -r ./spec-"$VERS"/proto/tendermint/** ./proto/tendermint
+# Get shortened ref of commit
+REF=$(curl -H "Accept: application/vnd.github.v3.sha" -qL \
+  "https://api.github.com/repos/tendermint/spec/commits/${VERS}" \
+  | cut -c -7)
+curl -qL "https://api.github.com/repos/tendermint/spec/${URL_PATH}${REF}" | tar -xzf - tendermint-spec-"$REF"/
 
-buf generate --path proto/tendermint
+cp -r ./tendermint-spec-"$REF"/proto/tendermint/* ./proto/tendermint
+cp -r ./tendermint-spec-"$REF"/third_party/** ./third_party
+
+MODNAME="$(go list -m)"
+find ./proto/tendermint -name '*.proto' -not -path "./proto/tendermint/abci/types.proto" \
+	-exec sh ./scripts/protopackage.sh {} "$MODNAME" ';'
+
```

Comment:
If I read this correctly, once `protopackage.sh` has been run, the input file could be deleted immediately, rather than having to re-find below:

```suggestion
find ./proto/tendermint -name '*.proto' -not -path "./proto/tendermint/abci/types.proto" \
	-exec sh ./scripts/protopackage.sh {} "$MODNAME" ';' -delete

```

Since you already filtered out `abci/types.proto` that one will not be harmed.

Reply:
Not sure I understand? We have only updated the file to contain the package name here, we haven't actually built the protos yet.

Reply:
Ah, sorry, of course. 🤦🏼‍♀️